pkg/cmd/label/create/create.go

```diff
@@ -0,0 +1,222 @@
+package create
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/label/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/pkg/prompt"
+	"github.com/spf13/cobra"
+)
+
+var randomColor = []string{
+	"B60205",
+	"D93F0B",
+	"FBCA04",
+	"0E8A16",
+	"006B75",
+	"1D76DB",
+	"0052CC",
+	"5319E7",
+	"E99695",
+	"F9D0C4",
+	"FEF2C0",
+	"C2E0C6",
+	"BFDADC",
+	"C5DEF5",
+	"BFD4F2",
+	"D4C5F9",
+}
+
+type CreateOptions struct {
+	HttpClient func() (*http.Client, error)
+	Config     func() (config.Config, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	Name        string
+	Description string
+	Color       string
+	Interactive bool
+}
+
+func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Command {
+	opts := CreateOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new label",
+		Example: heredoc.Doc(`
+		$ gh label create --name bug
+		$ gh label create --name bug --description "Something isn't working"
+		$ gh label create --name bug --description "Something isn't working" --color "E99695"
+	`),
+		Args: cobra.NoArgs,
+		RunE: func(c *cobra.Command, args []string) error {
+			nameProvided := c.Flags().Changed("name")
+			if !nameProvided {
+				opts.Interactive = true
+			}
+
+			if opts.Interactive && !opts.IO.CanPrompt() {
+				return cmdutil.FlagErrorf("must provide name when not running interactively")
+			}
+
+			opts.Name = strings.Trim(opts.Name, " ")
+
+			if runF != nil {
+				return runF(&opts)
+			}
+			return createRun(&opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Name, "name", "n", "", "Name of the label")
+	cmd.Flags().StringVarP(&opts.Description, "description", "d", "", "Description of the label")
+	cmd.Flags().StringVarP(&opts.Color, "color", "c", "", "Color of the label, if not specified will be random")
```

Comment:
I think we should allow the user to specify the hex code as `#FFFFFF` or `FFFFFF`, with or without the # symbol, and we just handle the trimming of the # symbol for them if it exists.